File: backend/src/modules/rtpengine/rtpengine.controller.ts
import { Controller, Get, Post, Body } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { RtpengineService } from './rtpengine.service';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('RTPEngine/NAT')
@Controller('api/rtpengine')
export class RtpengineController {
  constructor(private readonly rtpengineService: RtpengineService) {}

  @Get('show')
  @ApiOperation({ summary: '显示 RTPEngine 状态' })
  async showAll() {
    return ApiResponseDto.success(await this.rtpengineService.showAll());
  }

  @Post('reload')
  @ApiOperation({ summary: '重载 RTPEngine' })
  async reload() {
    await this.rtpengineService.reload();
    return ApiResponseDto.success(null, '重载成功');
  }

  @Post('enable')
  @ApiOperation({ summary: '启用/禁用 RTPEngine' })
  async enable(@Body() body: { url: string; flag: number }) {
    await this.rtpengineService.enable(body.url, body.flag);
    return ApiResponseDto.success(null, '操作成功');
  }

  @Post('ping')
  @ApiOperation({ summary: 'Ping RTPEngine' })
  async ping(@Body() body: { url: string }) {
    return ApiResponseDto.success(await this.rtpengineService.ping(body.url));
  }

  @Post('nat-ping')
  @ApiOperation({ summary: '启用/禁用 NAT Ping' })
  async enableNatPing(@Body() body: { flag: number }) {
    await this.rtpengineService.enableNatPing(body.flag);
    return ApiResponseDto.success(null, '操作成功');
  }

  @Get('status')
  @ApiOperation({ summary: '获取状态' })
  async getStatus() {
    return ApiResponseDto.success(await this.rtpengineService.getStatus());
  }
}

File: kam-admin-module-trace/src/main/java/io/kamailio/admin/modules/rtpengine/controller/RtpengineController.java
package io.kamailio.admin.modules.rtpengine.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.modules.rtpengine.service.RtpengineService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@Tag(name = "RTPEngine/NAT")
@RestController
@RequestMapping("/api/rtpengine")
public class RtpengineController {
    private final RtpengineService service;

    public RtpengineController(RtpengineService service) { this.service = service; }

    @GetMapping("/show")
    @Operation(summary = "显示 RTPEngine 状态")
    public ResponseEntity<ApiResponse<Object>> showAll() {
        return ResponseEntity.ok(ApiResponse.success(service.showAll()));
    }

    @PostMapping("/reload")
    @Operation(summary = "重载 RTPEngine")
    public ResponseEntity<ApiResponse<Void>> reload() {
        service.reload();
        return ResponseEntity.ok(ApiResponse.success(null, "重载成功"));
    }

    @PostMapping("/enable")
    @Operation(summary = "启用/禁用 RTPEngine")
    public ResponseEntity<ApiResponse<Void>> enable(@RequestBody Map<String, Object> body) {
        service.enable((String) body.get("url"), ((Number) body.get("flag")).intValue());
        return ResponseEntity.ok(ApiResponse.success(null, "操作成功"));
    }

    @PostMapping("/ping")
    @Operation(summary = "Ping RTPEngine")
    public ResponseEntity<ApiResponse<Object>> ping(@RequestBody Map<String, String> body) {
        return ResponseEntity.ok(ApiResponse.success(service.ping(body.get("url"))));
    }

    @PostMapping("/nat-ping")
    @Operation(summary = "启用/禁用 NAT Ping")
    public ResponseEntity<ApiResponse<Void>> enableNatPing(@RequestBody Map<String, Object> body) {
        int flag = body != null && body.get("flag") != null ? ((Number) body.get("flag")).intValue() : 0;
        service.enableNatPing(flag);
        return ResponseEntity.ok(ApiResponse.success(null, "操作成功"));
    }

    @GetMapping("/status")
    @Operation(summary = "获取状态")
    public ResponseEntity<ApiResponse<Map<String, Object>>> getStatus() {
        return ResponseEntity.ok(ApiResponse.success(service.getStatus()));
    }
}
